src/Microsoft.Health.Dicom.Core/Features/Store/StoreResponseBuilder.cs

```diff
@@ -64,6 +73,14 @@ public StoreResponse BuildResponse(string studyInstanceUid)
         return new StoreResponse(status, _dataset, _message);
     }
 
+    private bool LeniencyApplied()
+    {
+        DicomSequence referencedSOPSequence = _dataset.GetSequence(DicomTag.ReferencedSOPSequence);
+        return referencedSOPSequence
+            .Select(ds => ds.TryGetSequence(DicomTag.FailedAttributesSequence, out _) == true)
+            .Any();
```

Comment:
If  storeValidationResult.InvalidTagErrors has any items then we can set to partial success right?

Reply:
agree - we shouldn't need to evaluate the feature flag because previous versions will just skip this.